src/plugin/plugin.rs

```diff
@@ -14,21 +14,29 @@ pub type NoUserData = ();
 // Rapier physics engine.
 pub struct RapierPhysicsPlugin<PhysicsHooksData = ()> {
     physics_scale: f32,
+    system_setup: bool,
     _phantom: PhantomData<PhysicsHooksData>,
 }
 
-impl<PhysicsHooksData> RapierPhysicsPlugin<PhysicsHooksData> {
+impl<PhysicsHooksData: 'static + WorldQuery + Send + Sync> RapierPhysicsPlugin<PhysicsHooksData> {
     /// Specifies a scale ratio between the physics world and the bevy transforms.
     ///
     /// This affects the size of every elements in the physics engine, by multiplying
     /// all the length-related quantities by the `physics_scale` factor. This should
     /// likely always be 1.0 in 3D. In 2D, this is useful to specify a "pixels-per-meter"
     /// conversion ratio.
-    pub fn with_physics_scale(physics_scale: f32) -> Self {
-        Self {
-            physics_scale,
-            _phantom: PhantomData,
-        }
+    pub fn with_physics_scale(mut self, physics_scale: f32) -> Self {
```

Comment:
This method signature change is a breaking change.  I changed into the building pattern that most of bevy seems to use.  Any suggestions on how to configure options separately that are preferred are welcome :)